// @vapi-ai/server-sdk/dist/esm/api/resources/testSuiteRuns/client/Client.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as environments from "../../../../environments.mjs";
import * as core from "../../../../core/index.mjs";
import { mergeHeaders, mergeOnlyDefinedHeaders } from "../../../../core/headers.mjs";
import * as errors from "../../../../errors/index.mjs";
export class TestSuiteRuns {
    constructor(_options) {
        this._options = _options;
    }
    /**
     * @param {string} testSuiteId
     * @param {Vapi.TestSuiteRunControllerFindAllPaginatedRequest} request
     * @param {TestSuiteRuns.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @example
     *     await client.testSuiteRuns.testSuiteRunControllerFindAllPaginated("testSuiteId")
     */
    testSuiteRunControllerFindAllPaginated(testSuiteId, request = {}, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__testSuiteRunControllerFindAllPaginated(testSuiteId, request, requestOptions));
    }
    __testSuiteRunControllerFindAllPaginated(testSuiteId_1) {
        return __awaiter(this, arguments, void 0, function* (testSuiteId, request = {}, requestOptions) {
            var _a, _b, _c, _d, _e, _f;
            const { page, sortOrder, limit, createdAtGt, createdAtLt, createdAtGe, createdAtLe, updatedAtGt, updatedAtLt, updatedAtGe, updatedAtLe, } = request;
            const _queryParams = {};
            if (page !== undefined) {
                _queryParams["page"] = (_a = page === null || page === void 0 ? void 0 : page.toString()) !== null && _a !== void 0 ? _a : null;
            }
            if (sortOrder !== undefined) {
                _queryParams["sortOrder"] = sortOrder;
            }
            if (limit !== undefined) {
                _queryParams["limit"] = (_b = limit === null || limit === void 0 ? void 0 : limit.toString()) !== null && _b !== void 0 ? _b : null;
            }
            if (createdAtGt !== undefined) {
                _queryParams["createdAtGt"] = createdAtGt;
            }
            if (createdAtLt !== undefined) {
                _queryParams["createdAtLt"] = createdAtLt;
            }
            if (createdAtGe !== undefined) {
                _queryParams["createdAtGe"] = createdAtGe;
            }
            if (createdAtLe !== undefined) {
                _queryParams["createdAtLe"] = createdAtLe;
            }
            if (updatedAtGt !== undefined) {
                _queryParams["updatedAtGt"] = updatedAtGt;
            }
            if (updatedAtLt !== undefined) {
                _queryParams["updatedAtLt"] = updatedAtLt;
            }
            if (updatedAtGe !== undefined) {
                _queryParams["updatedAtGe"] = updatedAtGe;
            }
            if (updatedAtLe !== undefined) {
                _queryParams["updatedAtLe"] = updatedAtLe;
            }
            const _response = yield ((_c = this._options.fetcher) !== null && _c !== void 0 ? _c : core.fetcher)({
                url: core.url.join((_e = (_d = (yield core.Supplier.get(this._options.baseUrl))) !== null && _d !== void 0 ? _d : (yield core.Supplier.get(this._options.environment))) !== null && _e !== void 0 ? _e : environments.VapiEnvironment.Default, `test-suite/${encodeURIComponent(testSuiteId)}/run`),
                method: "GET",
                headers: mergeHeaders((_f = this._options) === null || _f === void 0 ? void 0 : _f.headers, mergeOnlyDefinedHeaders({ Authorization: yield this._getAuthorizationHeader() }), requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers),
                queryParameters: _queryParams,
                timeoutMs: (requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) != null ? requestOptions.timeoutInSeconds * 1000 : 60000,
                maxRetries: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                throw new errors.VapiError({
                    statusCode: _response.error.statusCode,
                    body: _response.error.body,
                    rawResponse: _response.rawResponse,
                });
            }
            switch (_response.error.reason) {
                case "non-json":
                    throw new errors.VapiError({
                        statusCode: _response.error.statusCode,
                        body: _response.error.rawBody,
                        rawResponse: _response.rawResponse,
                    });
                case "timeout":
                    throw new errors.VapiTimeoutError("Timeout exceeded when calling GET /test-suite/{testSuiteId}/run.");
                case "unknown":
                    throw new errors.VapiError({
                        message: _response.error.errorMessage,
                        rawResponse: _response.rawResponse,
                    });
            }
        });
    }
    /**
     * @param {string} testSuiteId
     * @param {Vapi.CreateTestSuiteRunDto} request
     * @param {TestSuiteRuns.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @example
     *     await client.testSuiteRuns.testSuiteRunControllerCreate("testSuiteId")
     */
    testSuiteRunControllerCreate(testSuiteId, request = {}, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__testSuiteRunControllerCreate(testSuiteId, request, requestOptions));
    }
    __testSuiteRunControllerCreate(testSuiteId_1) {
        return __awaiter(this, arguments, void 0, function* (testSuiteId, request = {}, requestOptions) {
            var _a, _b, _c, _d;
            const _response = yield ((_a = this._options.fetcher) !== null && _a !== void 0 ? _a : core.fetcher)({
                url: core.url.join((_c = (_b = (yield core.Supplier.get(this._options.baseUrl))) !== null && _b !== void 0 ? _b : (yield core.Supplier.get(this._options.environment))) !== null && _c !== void 0 ? _c : environments.VapiEnvironment.Default, `test-suite/${encodeURIComponent(testSuiteId)}/run`),
                method: "POST",
                headers: mergeHeaders((_d = this._options) === null || _d === void 0 ? void 0 : _d.headers, mergeOnlyDefinedHeaders({ Authorization: yield this._getAuthorizationHeader() }), requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers),
                contentType: "application/json",
                requestType: "json",
                body: request,
                timeoutMs: (requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) != null ? requestOptions.timeoutInSeconds * 1000 : 60000,
                maxRetries: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                throw new errors.VapiError({
                    statusCode: _response.error.statusCode,
                    body: _response.error.body,
                    rawResponse: _response.rawResponse,
                });
            }
            switch (_response.error.reason) {
                case "non-json":
                    throw new errors.VapiError({
                        statusCode: _response.error.statusCode,
                        body: _response.error.rawBody,
                        rawResponse: _response.rawResponse,
                    });
                case "timeout":
                    throw new errors.VapiTimeoutError("Timeout exceeded when calling POST /test-suite/{testSuiteId}/run.");
                case "unknown":
                    throw new errors.VapiError({
                        message: _response.error.errorMessage,
                        rawResponse: _response.rawResponse,
                    });
            }
        });
    }
    /**
     * @param {string} testSuiteId
     * @param {string} id
     * @param {TestSuiteRuns.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @example
     *     await client.testSuiteRuns.testSuiteRunControllerFindOne("testSuiteId", "id")
     */
    testSuiteRunControllerFindOne(testSuiteId, id, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__testSuiteRunControllerFindOne(testSuiteId, id, requestOptions));
    }
    __testSuiteRunControllerFindOne(testSuiteId, id, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d;
            const _response = yield ((_a = this._options.fetcher) !== null && _a !== void 0 ? _a : core.fetcher)({
                url: core.url.join((_c = (_b = (yield core.Supplier.get(this._options.baseUrl))) !== null && _b !== void 0 ? _b : (yield core.Supplier.get(this._options.environment))) !== null && _c !== void 0 ? _c : environments.VapiEnvironment.Default, `test-suite/${encodeURIComponent(testSuiteId)}/run/${encodeURIComponent(id)}`),
                method: "GET",
                headers: mergeHeaders((_d = this._options) === null || _d === void 0 ? void 0 : _d.headers, mergeOnlyDefinedHeaders({ Authorization: yield this._getAuthorizationHeader() }), requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers),
                timeoutMs: (requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) != null ? requestOptions.timeoutInSeconds * 1000 : 60000,
                maxRetries: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                throw new errors.VapiError({
                    statusCode: _response.error.statusCode,
                    body: _response.error.body,
                    rawResponse: _response.rawResponse,
                });
            }
            switch (_response.error.reason) {
                case "non-json":
                    throw new errors.VapiError({
                        statusCode: _response.error.statusCode,
                        body: _response.error.rawBody,
                        rawResponse: _response.rawResponse,
                    });
                case "timeout":
                    throw new errors.VapiTimeoutError("Timeout exceeded when calling GET /test-suite/{testSuiteId}/run/{id}.");
                case "unknown":
                    throw new errors.VapiError({
                        message: _response.error.errorMessage,
                        rawResponse: _response.rawResponse,
                    });
            }
        });
    }
    /**
     * @param {string} testSuiteId
     * @param {string} id
     * @param {TestSuiteRuns.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @example
     *     await client.testSuiteRuns.testSuiteRunControllerRemove("testSuiteId", "id")
     */
    testSuiteRunControllerRemove(testSuiteId, id, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__testSuiteRunControllerRemove(testSuiteId, id, requestOptions));
    }
    __testSuiteRunControllerRemove(testSuiteId, id, requestOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c, _d;
            const _response = yield ((_a = this._options.fetcher) !== null && _a !== void 0 ? _a : core.fetcher)({
                url: core.url.join((_c = (_b = (yield core.Supplier.get(this._options.baseUrl))) !== null && _b !== void 0 ? _b : (yield core.Supplier.get(this._options.environment))) !== null && _c !== void 0 ? _c : environments.VapiEnvironment.Default, `test-suite/${encodeURIComponent(testSuiteId)}/run/${encodeURIComponent(id)}`),
                method: "DELETE",
                headers: mergeHeaders((_d = this._options) === null || _d === void 0 ? void 0 : _d.headers, mergeOnlyDefinedHeaders({ Authorization: yield this._getAuthorizationHeader() }), requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers),
                timeoutMs: (requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) != null ? requestOptions.timeoutInSeconds * 1000 : 60000,
                maxRetries: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                throw new errors.VapiError({
                    statusCode: _response.error.statusCode,
                    body: _response.error.body,
                    rawResponse: _response.rawResponse,
                });
            }
            switch (_response.error.reason) {
                case "non-json":
                    throw new errors.VapiError({
                        statusCode: _response.error.statusCode,
                        body: _response.error.rawBody,
                        rawResponse: _response.rawResponse,
                    });
                case "timeout":
                    throw new errors.VapiTimeoutError("Timeout exceeded when calling DELETE /test-suite/{testSuiteId}/run/{id}.");
                case "unknown":
                    throw new errors.VapiError({
                        message: _response.error.errorMessage,
                        rawResponse: _response.rawResponse,
                    });
            }
        });
    }
    /**
     * @param {string} testSuiteId
     * @param {string} id
     * @param {Vapi.UpdateTestSuiteRunDto} request
     * @param {TestSuiteRuns.RequestOptions} requestOptions - Request-specific configuration.
     *
     * @example
     *     await client.testSuiteRuns.testSuiteRunControllerUpdate("testSuiteId", "id")
     */
    testSuiteRunControllerUpdate(testSuiteId, id, request = {}, requestOptions) {
        return core.HttpResponsePromise.fromPromise(this.__testSuiteRunControllerUpdate(testSuiteId, id, request, requestOptions));
    }
    __testSuiteRunControllerUpdate(testSuiteId_1, id_1) {
        return __awaiter(this, arguments, void 0, function* (testSuiteId, id, request = {}, requestOptions) {
            var _a, _b, _c, _d;
            const _response = yield ((_a = this._options.fetcher) !== null && _a !== void 0 ? _a : core.fetcher)({
                url: core.url.join((_c = (_b = (yield core.Supplier.get(this._options.baseUrl))) !== null && _b !== void 0 ? _b : (yield core.Supplier.get(this._options.environment))) !== null && _c !== void 0 ? _c : environments.VapiEnvironment.Default, `test-suite/${encodeURIComponent(testSuiteId)}/run/${encodeURIComponent(id)}`),
                method: "PATCH",
                headers: mergeHeaders((_d = this._options) === null || _d === void 0 ? void 0 : _d.headers, mergeOnlyDefinedHeaders({ Authorization: yield this._getAuthorizationHeader() }), requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.headers),
                contentType: "application/json",
                requestType: "json",
                body: request,
                timeoutMs: (requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.timeoutInSeconds) != null ? requestOptions.timeoutInSeconds * 1000 : 60000,
                maxRetries: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.maxRetries,
                abortSignal: requestOptions === null || requestOptions === void 0 ? void 0 : requestOptions.abortSignal,
            });
            if (_response.ok) {
                return { data: _response.body, rawResponse: _response.rawResponse };
            }
            if (_response.error.reason === "status-code") {
                throw new errors.VapiError({
                    statusCode: _response.error.statusCode,
                    body: _response.error.body,
                    rawResponse: _response.rawResponse,
                });
            }
            switch (_response.error.reason) {
                case "non-json":
                    throw new errors.VapiError({
                        statusCode: _response.error.statusCode,
                        body: _response.error.rawBody,
                        rawResponse: _response.rawResponse,
                    });
                case "timeout":
                    throw new errors.VapiTimeoutError("Timeout exceeded when calling PATCH /test-suite/{testSuiteId}/run/{id}.");
                case "unknown":
                    throw new errors.VapiError({
                        message: _response.error.errorMessage,
                        rawResponse: _response.rawResponse,
                    });
            }
        });
    }
    _getAuthorizationHeader() {
        return __awaiter(this, void 0, void 0, function* () {
            return `Bearer ${yield core.Supplier.get(this._options.token)}`;
        });
    }
}
