// @vapi-ai/server-sdk/dist/cjs/api/types/AnalyticsOperation.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsOperation = void 0;
var AnalyticsOperation;
(function (AnalyticsOperation) {
    AnalyticsOperation.Operation = {
        Sum: "sum",
        Avg: "avg",
        Count: "count",
        Min: "min",
        Max: "max",
        History: "history",
    };
    AnalyticsOperation.Column = {
        Id: "id",
        Cost: "cost",
        CostBreakdownLlm: "costBreakdown.llm",
        CostBreakdownStt: "costBreakdown.stt",
        CostBreakdownTts: "costBreakdown.tts",
        CostBreakdownVapi: "costBreakdown.vapi",
        CostBreakdownTtsCharacters: "costBreakdown.ttsCharacters",
        CostBreakdownLlmPromptTokens: "costBreakdown.llmPromptTokens",
        CostBreakdownLlmCompletionTokens: "costBreakdown.llmCompletionTokens",
        Duration: "duration",
        Concurrency: "concurrency",
        MinutesUsed: "minutesUsed",
    };
})(AnalyticsOperation || (exports.AnalyticsOperation = AnalyticsOperation = {}));
