// @vapi-ai/server-sdk/dist/cjs/api/types/AssistantOverrides.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssistantOverrides = void 0;
var AssistantOverrides;
(function (AssistantOverrides) {
    AssistantOverrides.FirstMessageMode = {
        AssistantSpeaksFirst: "assistant-speaks-first",
        AssistantSpeaksFirstWithModelGeneratedMessage: "assistant-speaks-first-with-model-generated-message",
        AssistantWaitsForUser: "assistant-waits-for-user",
    };
    let ClientMessages;
    (function (ClientMessages) {
        ClientMessages.Item = {
            ConversationUpdate: "conversation-update",
            FunctionCall: "function-call",
            FunctionCallResult: "function-call-result",
            Hang: "hang",
            LanguageChanged: "language-changed",
            Metadata: "metadata",
            ModelOutput: "model-output",
            SpeechUpdate: "speech-update",
            StatusUpdate: "status-update",
            Transcript: "transcript",
            ToolCalls: "tool-calls",
            ToolCallsResult: "tool-calls-result",
            ToolCompleted: "tool.completed",
            TransferUpdate: "transfer-update",
            UserInterrupted: "user-interrupted",
            VoiceInput: "voice-input",
            WorkflowNodeStarted: "workflow.node.started",
        };
    })(ClientMessages = AssistantOverrides.ClientMessages || (AssistantOverrides.ClientMessages = {}));
    let ServerMessages;
    (function (ServerMessages) {
        ServerMessages.Item = {
            ConversationUpdate: "conversation-update",
            EndOfCallReport: "end-of-call-report",
            FunctionCall: "function-call",
            Hang: "hang",
            LanguageChanged: "language-changed",
            LanguageChangeDetected: "language-change-detected",
            ModelOutput: "model-output",
            PhoneCallControl: "phone-call-control",
            SpeechUpdate: "speech-update",
            StatusUpdate: "status-update",
            Transcript: "transcript",
            TranscriptTranscriptTypeFinal: 'transcript[transcriptType="final"]',
            ToolCalls: "tool-calls",
            TransferDestinationRequest: "transfer-destination-request",
            TransferUpdate: "transfer-update",
            UserInterrupted: "user-interrupted",
            VoiceInput: "voice-input",
        };
    })(ServerMessages = AssistantOverrides.ServerMessages || (AssistantOverrides.ServerMessages = {}));
})(AssistantOverrides || (exports.AssistantOverrides = AssistantOverrides = {}));
