// @vapi-ai/server-sdk/dist/cjs/api/types/DeepgramVoice.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeepgramVoice = void 0;
var DeepgramVoice;
(function (DeepgramVoice) {
    DeepgramVoice.VoiceId = {
        Asteria: "asteria",
        Luna: "luna",
        Stella: "stella",
        Athena: "athena",
        Hera: "hera",
        Orion: "orion",
        Arcas: "arcas",
        Perseus: "perseus",
        Angus: "angus",
        Orpheus: "orpheus",
        Helios: "helios",
        Zeus: "zeus",
        Thalia: "thalia",
        Andromeda: "andromeda",
        Helena: "helena",
        Apollo: "apollo",
        Aries: "aries",
        Amalthea: "amalthea",
        Atlas: "atlas",
        Aurora: "aurora",
        Callista: "callista",
        Cora: "cora",
        Cordelia: "cordelia",
        Delia: "delia",
        Draco: "draco",
        Electra: "electra",
        Harmonia: "harmonia",
        Hermes: "hermes",
        Hyperion: "hyperion",
        Iris: "iris",
        Janus: "janus",
        Juno: "juno",
        Jupiter: "jupiter",
        Mars: "mars",
        Minerva: "minerva",
        Neptune: "neptune",
        Odysseus: "odysseus",
        Ophelia: "ophelia",
        Pandora: "pandora",
        Phoebe: "phoebe",
        Pluto: "pluto",
        Saturn: "saturn",
        Selene: "selene",
        Theia: "theia",
        Vesta: "vesta",
    };
    DeepgramVoice.Model = {
        Aura: "aura",
        Aura2: "aura-2",
    };
})(DeepgramVoice || (exports.DeepgramVoice = DeepgramVoice = {}));
