// @vapi-ai/server-sdk/dist/cjs/api/types/FallbackInworldVoice.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FallbackInworldVoice = void 0;
var FallbackInworldVoice;
(function (FallbackInworldVoice) {
    FallbackInworldVoice.VoiceId = {
        Alex: "Alex",
        Ashley: "Ashley",
        Craig: "Craig",
        Deborah: "Deborah",
        Dennis: "Dennis",
        Edward: "Edward",
        Elizabeth: "Elizabeth",
        Hades: "Hades",
        Julia: "Julia",
        Pixie: "Pixie",
        Mark: "Mark",
        Olivia: "Olivia",
        Priya: "Priya",
        Ronald: "Ronald",
        Sarah: "Sarah",
        Shaun: "Shaun",
        Theodore: "Theodore",
        Timothy: "Timothy",
        Wendy: "Wendy",
        Dominus: "Dominus",
        Yichen: "Yichen",
        Xiaoyin: "Xiaoyin",
        Xinyi: "Xinyi",
        Jing: "Jing",
        Erik: "Erik",
        Katrien: "Katrien",
        Lennart: "Lennart",
        Lore: "Lore",
        Alain: "Alain",
        Helene: "H\u00E9l\u00E8ne",
        Mathieu: "Mathieu",
        Etienne: "\u00C9tienne",
        Johanna: "Johanna",
        Josef: "Josef",
        Gianni: "Gianni",
        Orietta: "Orietta",
        Asuka: "Asuka",
        Satoshi: "Satoshi",
        Hyunwoo: "Hyunwoo",
        Minji: "Minji",
        Seojun: "Seojun",
        Yoona: "Yoona",
        Szymon: "Szymon",
        Wojciech: "Wojciech",
        Heitor: "Heitor",
        Maite: "Mait\u00EA",
        Diego: "Diego",
        Lupita: "Lupita",
        Miguel: "Miguel",
        Rafael: "Rafael",
    };
    FallbackInworldVoice.LanguageCode = {
        En: "en",
        Zh: "zh",
        Ko: "ko",
        Nl: "nl",
        Fr: "fr",
        Es: "es",
        Ja: "ja",
        De: "de",
        It: "it",
        Pl: "pl",
        Pt: "pt",
    };
})(FallbackInworldVoice || (exports.FallbackInworldVoice = FallbackInworldVoice = {}));
