// @vapi-ai/server-sdk/dist/cjs/api/types/FallbackPlayHtVoice.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FallbackPlayHtVoice = void 0;
var FallbackPlayHtVoice;
(function (FallbackPlayHtVoice) {
    FallbackPlayHtVoice.Emotion = {
        FemaleHappy: "female_happy",
        FemaleSad: "female_sad",
        FemaleAngry: "female_angry",
        FemaleFearful: "female_fearful",
        FemaleDisgust: "female_disgust",
        FemaleSurprised: "female_surprised",
        MaleHappy: "male_happy",
        MaleSad: "male_sad",
        MaleAngry: "male_angry",
        MaleFearful: "male_fearful",
        MaleDisgust: "male_disgust",
        MaleSurprised: "male_surprised",
    };
    FallbackPlayHtVoice.Model = {
        PlayHt20: "PlayHT2.0",
        PlayHt20Turbo: "PlayHT2.0-turbo",
        Play30Mini: "Play3.0-mini",
        PlayDialog: "PlayDialog",
    };
    FallbackPlayHtVoice.Language = {
        Afrikaans: "afrikaans",
        Albanian: "albanian",
        Amharic: "amharic",
        Arabic: "arabic",
        Bengali: "bengali",
        Bulgarian: "bulgarian",
        Catalan: "catalan",
        Croatian: "croatian",
        Czech: "czech",
        Danish: "danish",
        Dutch: "dutch",
        English: "english",
        French: "french",
        Galician: "galician",
        German: "german",
        Greek: "greek",
        Hebrew: "hebrew",
        Hindi: "hindi",
        Hungarian: "hungarian",
        Indonesian: "indonesian",
        Italian: "italian",
        Japanese: "japanese",
        Korean: "korean",
        Malay: "malay",
        Mandarin: "mandarin",
        Polish: "polish",
        Portuguese: "portuguese",
        Russian: "russian",
        Serbian: "serbian",
        Spanish: "spanish",
        Swedish: "swedish",
        Tagalog: "tagalog",
        Thai: "thai",
        Turkish: "turkish",
        Ukrainian: "ukrainian",
        Urdu: "urdu",
        Xhosa: "xhosa",
    };
})(FallbackPlayHtVoice || (exports.FallbackPlayHtVoice = FallbackPlayHtVoice = {}));
