// @vapi-ai/server-sdk/dist/cjs/api/types/FormatPlan.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormatPlan = void 0;
var FormatPlan;
(function (FormatPlan) {
    let FormattersEnabled;
    (function (FormattersEnabled) {
        FormattersEnabled.Item = {
            Markdown: "markdown",
            Asterisk: "asterisk",
            Quote: "quote",
            Dash: "dash",
            Newline: "newline",
            Colon: "colon",
            Acronym: "acronym",
            DollarAmount: "dollarAmount",
            Email: "email",
            Date: "date",
            Time: "time",
            Distance: "distance",
            Unit: "unit",
            Percentage: "percentage",
            PhoneNumber: "phoneNumber",
            Number: "number",
            StripAsterisk: "stripAsterisk",
        };
    })(FormattersEnabled = FormatPlan.FormattersEnabled || (FormatPlan.FormattersEnabled = {}));
})(FormatPlan || (exports.FormatPlan = FormatPlan = {}));
