// @vapi-ai/server-sdk/dist/cjs/api/types/SyncVoiceLibraryDto.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncVoiceLibraryDto = void 0;
var SyncVoiceLibraryDto;
(function (SyncVoiceLibraryDto) {
    let Providers;
    (function (Providers) {
        Providers.Item = {
            Vapi: "vapi",
            ElevenLabs: "11labs",
            Azure: "azure",
            Cartesia: "cartesia",
            CustomVoice: "custom-voice",
            Deepgram: "deepgram",
            Hume: "hume",
            Lmnt: "lmnt",
            Neuphonic: "neuphonic",
            Openai: "openai",
            Playht: "playht",
            RimeAi: "rime-ai",
            SmallestAi: "smallest-ai",
            Tavus: "tavus",
            Sesame: "sesame",
            Inworld: "inworld",
        };
    })(Providers = SyncVoiceLibraryDto.Providers || (SyncVoiceLibraryDto.Providers = {}));
})(SyncVoiceLibraryDto || (exports.SyncVoiceLibraryDto = SyncVoiceLibraryDto = {}));
