// @vapi-ai/server-sdk/dist/cjs/api/types/UpdateAzureCredentialDto.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAzureCredentialDto = void 0;
var UpdateAzureCredentialDto;
(function (UpdateAzureCredentialDto) {
    UpdateAzureCredentialDto.Service = {
        Speech: "speech",
        BlobStorage: "blob_storage",
    };
    UpdateAzureCredentialDto.Region = {
        Australia: "australia",
        Canadaeast: "canadaeast",
        Canadacentral: "canadacentral",
        Eastus2: "eastus2",
        Eastus: "eastus",
        France: "france",
        India: "india",
        Japaneast: "japaneast",
        Japanwest: "japanwest",
        Uaenorth: "uaenorth",
        Northcentralus: "northcentralus",
        Norway: "norway",
        Southcentralus: "southcentralus",
        Swedencentral: "swedencentral",
        Switzerland: "switzerland",
        Uk: "uk",
        Westus: "westus",
        Westus3: "westus3",
    };
})(UpdateAzureCredentialDto || (exports.UpdateAzureCredentialDto = UpdateAzureCredentialDto = {}));
