// @vapi-ai/server-sdk/dist/esm/api/types/GoogleTranscriber.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
export var GoogleTranscriber;
(function (GoogleTranscriber) {
    GoogleTranscriber.Model = {
        Gemini25Pro: "gemini-2.5-pro",
        Gemini25Flash: "gemini-2.5-flash",
        Gemini25FlashLite: "gemini-2.5-flash-lite",
        Gemini25ProPreview0506: "gemini-2.5-pro-preview-05-06",
        Gemini25FlashPreview0520: "gemini-2.5-flash-preview-05-20",
        Gemini25FlashPreview0417: "gemini-2.5-flash-preview-04-17",
        Gemini20FlashThinkingExp: "gemini-2.0-flash-thinking-exp",
        Gemini20ProExp0205: "gemini-2.0-pro-exp-02-05",
        Gemini20Flash: "gemini-2.0-flash",
        Gemini20FlashLite: "gemini-2.0-flash-lite",
        Gemini20FlashLitePreview0205: "gemini-2.0-flash-lite-preview-02-05",
        Gemini20FlashExp: "gemini-2.0-flash-exp",
        Gemini20FlashRealtimeExp: "gemini-2.0-flash-realtime-exp",
        Gemini15Flash: "gemini-1.5-flash",
        Gemini15Flash002: "gemini-1.5-flash-002",
        Gemini15Pro: "gemini-1.5-pro",
        Gemini15Pro002: "gemini-1.5-pro-002",
        Gemini10Pro: "gemini-1.0-pro",
    };
    GoogleTranscriber.Language = {
        Multilingual: "Multilingual",
        Arabic: "Arabic",
        Bengali: "Bengali",
        Bulgarian: "Bulgarian",
        Chinese: "Chinese",
        Croatian: "Croatian",
        Czech: "Czech",
        Danish: "Danish",
        Dutch: "Dutch",
        English: "English",
        Estonian: "Estonian",
        Finnish: "Finnish",
        French: "French",
        German: "German",
        Greek: "Greek",
        Hebrew: "Hebrew",
        Hindi: "Hindi",
        Hungarian: "Hungarian",
        Indonesian: "Indonesian",
        Italian: "Italian",
        Japanese: "Japanese",
        Korean: "Korean",
        Latvian: "Latvian",
        Lithuanian: "Lithuanian",
        Norwegian: "Norwegian",
        Polish: "Polish",
        Portuguese: "Portuguese",
        Romanian: "Romanian",
        Russian: "Russian",
        Serbian: "Serbian",
        Slovak: "Slovak",
        Slovenian: "Slovenian",
        Spanish: "Spanish",
        Swahili: "Swahili",
        Swedish: "Swedish",
        Thai: "Thai",
        Turkish: "Turkish",
        Ukrainian: "Ukrainian",
        Vietnamese: "Vietnamese",
    };
})(GoogleTranscriber || (GoogleTranscriber = {}));
