// @vapi-ai/server-sdk/dist/cjs/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.analytics = exports.squads = exports.workflow = exports.files = exports.assistants = exports.logs = exports.testSuiteRuns = exports.testSuiteTests = exports.testSuites = exports.knowledgeBases = exports.tools = exports.phoneNumbers = exports.sessions = exports.campaigns = exports.chats = exports.calls = void 0;
exports.calls = __importStar(require("./calls/index.js"));
__exportStar(require("./calls/types/index.js"), exports);
exports.chats = __importStar(require("./chats/index.js"));
__exportStar(require("./chats/types/index.js"), exports);
exports.campaigns = __importStar(require("./campaigns/index.js"));
__exportStar(require("./campaigns/types/index.js"), exports);
exports.sessions = __importStar(require("./sessions/index.js"));
__exportStar(require("./sessions/types/index.js"), exports);
exports.phoneNumbers = __importStar(require("./phoneNumbers/index.js"));
__exportStar(require("./phoneNumbers/types/index.js"), exports);
exports.tools = __importStar(require("./tools/index.js"));
__exportStar(require("./tools/types/index.js"), exports);
exports.knowledgeBases = __importStar(require("./knowledgeBases/index.js"));
__exportStar(require("./knowledgeBases/types/index.js"), exports);
exports.testSuites = __importStar(require("./testSuites/index.js"));
__exportStar(require("./testSuites/types/index.js"), exports);
exports.testSuiteTests = __importStar(require("./testSuiteTests/index.js"));
__exportStar(require("./testSuiteTests/types/index.js"), exports);
exports.testSuiteRuns = __importStar(require("./testSuiteRuns/index.js"));
__exportStar(require("./testSuiteRuns/types/index.js"), exports);
exports.logs = __importStar(require("./logs/index.js"));
__exportStar(require("./logs/types/index.js"), exports);
exports.assistants = __importStar(require("./assistants/index.js"));
exports.files = __importStar(require("./files/index.js"));
exports.workflow = __importStar(require("./workflow/index.js"));
exports.squads = __importStar(require("./squads/index.js"));
exports.analytics = __importStar(require("./analytics/index.js"));
__exportStar(require("./calls/client/requests/index.js"), exports);
__exportStar(require("./chats/client/requests/index.js"), exports);
__exportStar(require("./campaigns/client/requests/index.js"), exports);
__exportStar(require("./sessions/client/requests/index.js"), exports);
__exportStar(require("./assistants/client/requests/index.js"), exports);
__exportStar(require("./phoneNumbers/client/requests/index.js"), exports);
__exportStar(require("./tools/client/requests/index.js"), exports);
__exportStar(require("./files/client/requests/index.js"), exports);
__exportStar(require("./knowledgeBases/client/requests/index.js"), exports);
__exportStar(require("./workflow/client/requests/index.js"), exports);
__exportStar(require("./squads/client/requests/index.js"), exports);
__exportStar(require("./testSuites/client/requests/index.js"), exports);
__exportStar(require("./testSuiteTests/client/requests/index.js"), exports);
__exportStar(require("./testSuiteRuns/client/requests/index.js"), exports);
__exportStar(require("./analytics/client/requests/index.js"), exports);
__exportStar(require("./logs/client/requests/index.js"), exports);
